tests/PHPStan/Rules/Comparison/data/bug-12755.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types = 1);
+
+namespace Bug12755;
+
+class HelloWorld
+{
+	/**
+	 * @param array{
+	 *     key1: ?int,
+	 *     key2: ?string,
+	 * } $myArray
+	 */
+	public function testOther(array $myArray): ?\stdClass
+	{
+	    if (\in_array(null, $myArray, true)) {
+	        return null;
+	    }
+
+		return (object) $myArray;
+	}
+		
+	/**
+	 * @param array{
+	 *     key1: ?bool,
+	 * } $myArray
+	 */
+	public function testBool(array $myArray): ?\stdClass
+	{
+	    if (\in_array(null, $myArray, true)) {
+	        return null;
+	    }
+
+		return (object) $myArray;
+	}
```

Comment:
you should test a case which is expected to still emit an error after the change (a array which always contains null)